ui/src/app/applications/components/application-details/application-resource-filter.tsx

```diff
@@ -0,0 +1,114 @@
+
+    const ResourceFilter = (p: {label: string; prefix: string; options: string[]; field?: boolean; radio?: boolean}) => {
+        return loading ? (
+            <div>Loading...</div>
+        ) : (
+            <Filter
+                label={p.label}
+                selected={selectedFor(p.prefix)}
+                setSelected={v => setFilters(p.prefix, v)}
+                options={p.options.map(label => {
+                    return {label};
+                })}
+                field={!!p.field}
+                radio={!!p.radio}
+            />
+        );
+    };
+
+    // we need to include ones that might have been filter in other apps that do not apply to the current app,
+    // otherwise the user will not be able to clear them from this panel
+    const alreadyFilteredOn = (prefix: string) => resourceFilter.filter(f => f.startsWith(prefix + ':')).map(removePrefix(prefix));
+
+    const kinds = tree.nodes
+        .map(x => x.kind)
+        .concat(alreadyFilteredOn('kind'))
+        .filter(uniq)
+        .sort();
+    const namespaces = tree.nodes
+        .map(x => x.namespace)
+        .concat(alreadyFilteredOn('namespace'))
+        .filter(uniq)
+        .sort();
+
+    const selectedFor = (prefix: string) => {
+        return groupedFilters[prefix] ? groupedFilters[prefix].split(',').map(removePrefix(prefix)) : [];
+    };
+
+    return (
+        <>
+            <div className='applications-list__filters__title'>
+                FILTERS <i className='fa fa-filter' />
+                {pref.resourceFilter.length > 0 && (
+                    <ActionButton label={'CLEAR ALL'} action={() => onClearFilter()} style={{marginLeft: 'auto', fontSize: '12px', lineHeight: '5px', display: 'block'}} />
+                )}
+                <ActionButton
+                    label={!shown ? 'SHOW' : 'HIDE'}
+                    action={() => setShown(!shown)}
+                    style={{marginLeft: pref.resourceFilter.length > 0 ? '5px' : 'auto', fontSize: '12px', lineHeight: '5px', display: !shown && 'block'}}
+                />
+            </div>
+            {shown && (
+                <div className='applications-list__filters'>
+                    {ResourceFilter({label: 'OWNERSHIP', prefix: 'ownership', options: ['Owners', 'Owned']})}
+                    {ResourceFilter({label: 'SYNC STATUS', prefix: 'sync', options: ['Synced', 'OutOfSync']})}
+                    {ResourceFilter({label: 'HEALTH STATUS', prefix: 'health', options: ['Healthy', 'Progressing', 'Degraded', 'Suspended', 'Missing', 'Unknown']})}
+                    {ResourceFilter({label: 'KINDS', prefix: 'kind', options: kinds, field: true})}
+                    {namespaces.length > 1 && ResourceFilter({label: 'NAMESPACES', prefix: 'namespace', options: (namespaces || []).filter(l => l && l !== ''), field: true})}
+                    {ResourceFilter({label: 'CREATED WITHIN', prefix: 'createdWithin', options: ['1m', '3m', '5m', '15m', '60m'], radio: true})}
```

Comment:
Let rename `CREATED WITHIN` to age please

Reply:
That's a great idea, `CREATED WITHIN` was long and pushed the clear button to a separate row